mockOSApi.v2/Services/AuthenticationService.cs

```diff
@@ -0,0 +1,133 @@
+    {
+        _userRepositoryKv = userRepositoryKv;
+        _configuration = configuration;
+        _mapper = mapper;
+    }
+
+    public async Task<User?> AuthenticateUser(string username, string password)
+    {
+        // allow anonymous authentication
+        if (username == "anonymous")
+        {
+            return new User
+            {
+                Username = username,
+                Role = Roles.User
+            };
+        }
+        if (!_userRepositoryKv.UserExists(username))
+        {
+            return null;
+        }
+        var user = await _userRepositoryKv.GetUser(username);
+
+        if (user.Password == password)
+        {
+            return user;
+        }
+        return null;
+    }
+
+    public string GenerateToken(User user)
+    {
+        var tokenHandler = new JwtSecurityTokenHandler();
+        var key = Encoding.ASCII.GetBytes(_configuration["SecretString"]);
+
+        var tokenDescriptor = new SecurityTokenDescriptor // constructing the token descriptor
+        {
+            Subject = new ClaimsIdentity(new Claim[]
+           {
+                new Claim(ClaimTypes.Name, user.Username.ToString())
+               }),
+            Expires = DateTime.UtcNow.AddMinutes(30),
+            SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key), SecurityAlgorithms.HmacSha256Signature)
+        };
+
+        var token = tokenHandler.CreateToken(tokenDescriptor);
+        return tokenHandler.WriteToken(token);
+
+    }
+
+    public bool IsUserAuthorized(string? tokenString)
+    {
+        if (tokenString == null)
+            return false;
+        var token = new JwtSecurityTokenHandler().ReadJwtToken(tokenString);
+        var username = token.Claims.FirstOrDefault(c => c.ValueType == ClaimValueTypes.String).Value;
+        if (username == null) { return false; }
+        // get username from token
+        if (token.ValidTo > DateTime.Now)
+        { // check if token is not expired
+            return false;
+        }
+        return true;
+        // more checks to be performed
+    }
+
+    public User? GetUser(string? tokenString)
+    {
+        // checks should be already made by IsUserAuthorized
+        var token = new JwtSecurityTokenHandler().ReadJwtToken(tokenString);
+        var username = token.Claims.FirstOrDefault(c => c.ValueType == ClaimValueTypes.String).Value;
+        var user = new User
+        {
+            Username = username,
+            Role = Roles.Administrator,
+        };
+        return user;
+    }
+
+    public async Task<UserDTO?> RegisterUser(UserCreationDTO userTemplate)
+    {
+        var user = _mapper.Map<User>(userTemplate);
+        // will implement password policy checks later
+        try
+        {
+            var inUse = await _userRepositoryKv.GetUser(user.Username);
+        }
+        catch (Azure.RequestFailedException e)
```

Comment:
This is kinda bad practice. You should check if the secret exists and return true/false instead of catching an exact exception.
If the azure library u are depending on changes tomorrow this code will break